src/runtime/CollectionWrappers/SequenceWrapper.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class SequenceWrapper<T> : IterableWrapper<T>, ICollection<T>
+    {
+        public SequenceWrapper(PyObject pyObj) : base(pyObj)
+        {
+
+        }
+
+        public int Count
+        {
+            get
+            {
+                var size = Runtime.PySequence_Size(pyObject.Handle);
+                if (size == -1)
+                {
+                    Runtime.CheckExceptionOccurred();
+                }
+
+                return (int)size;
+            }
+        }
+
+        public virtual bool IsReadOnly => false;
+
+        public virtual void Add(T item)
+        {
+            //not implemented for Python sequence.
+            //ICollection is the closest analogue but it doesn't map perfectly.
+            //SequenceWrapper is not final and can be subclassed if necessary
+            throw new NotImplementedException();
+        }
+
+        public void Clear()
+        {
+            if (IsReadOnly)
+                throw new NotImplementedException();
+            var result = Runtime.PySequence_DelSlice(pyObject.Handle, 0, Count);
+            if (result == -1)
+            {
+                Runtime.CheckExceptionOccurred();
+            }
+        }
+
+        public bool Contains(T item)
+        {
+            //not sure if IEquatable is implemented and this will work!
+            foreach (var element in this)
+                if (element.Equals(item)) return true;
+
+            return false;
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            var index = 0;
```

Comment:
Please, check inputs, and, if possible that the collection will fit before making an attempt to copy.

Generally for any operation it is highly preferred that it either completes successfully, or does not make any changes to its inputs in case of exception.